src/map/status.c

```diff
@@ -8807,29 +8856,29 @@ int status_change_start(struct block_list *src, struct block_list *bl, enum sc_t
 				break;
 			case SC_WEAPONBLOCKING:
 				val2 = 10 + 2 * val1; // Chance
-				val4 = tick / 5000;
 				tick_time = 5000; // [GodLesZ] tick time
+				val4 = tick / tick_time;
 				break;
 			case SC_TOXIN:
-				val4 = tick / 10000;
 				tick_time = 10000; // [GodLesZ] tick time
+				val4 = tick / tick_time;
 				break;
 			case SC_MAGICMUSHROOM:
-				val4 = tick / 4000;
 				tick_time = 4000; // [GodLesZ] tick time
+				val4 = tick / tick_time;
 				break;
 			case SC_PYREXIA:
-				status->change_start(src, bl,SC_BLIND,10000,val1,0,0,0,30000,SCFLAG_NOAVOID|SCFLAG_FIXEDTICK|SCFLAG_FIXEDRATE); // Blind status that last for 30 seconds
-				val4 = tick / 3000;
 				tick_time = 3000; // [GodLesZ] tick time
+				val4 = tick / tick_time;
+				status->change_start(src, bl,SC_BLIND,10000,val1,0,0,0,30000,SCFLAG_NOAVOID|SCFLAG_FIXEDTICK|SCFLAG_FIXEDRATE); // Blind status that last for 30 seconds
```

Comment:
This still needs work, the blind and hallucination statuses should be unremovable, hallucination it causes isn't normal one that flips the screen but only changes damage numbers. There's also some strange additional effect (taken from iROwiki): "If the affected entity takes damage while under the effects of this poison, it will remain in flinching motion, which will interrupt the skill casting".